spec/presenters/two_factor_auth_code/phoneable.spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'rails_helper'
+
+phone_update_path = 'a/link/to/update'
+
+describe TwoFactorAuthCode::Phoneable do
+  let(:presenter) do
+    TwoFactorAuthCode::PhoneDeliveryPresenter.new(attributes_for(:generic_otp_presenter))
+  end
+
+  let(:unconfirmed) do
+    data = attributes_for(:generic_otp_presenter).merge!(
+      reenter_phone_number_path: phone_update_path,
+      unconfirmed_phone: true
+    )
+
+    TwoFactorAuthCode::PhoneDeliveryPresenter.new(data)
+  end
+
+  describe '#phone_fallback_link' do
+    it 'returns a link for voice delivery if user initially selected sms' do
+      output = presenter.phone_fallback_link
+
+      expect(output).to include(t('links.two_factor_authentication.voice'))
+      expect(output).to match(/<a href/)
+    end
+
+    it 'return a link for sms delivery if user initially selected voice' do
+      output = presenter.phone_fallback_link
+
+      expect(output).to include(t('links.two_factor_authentication.sms'))
+    end
```

Comment:
Yeah, I don't think it's great either. An alternative is to move this test to a view spec, where we can use the Rails URL helpers and not have to hardcode these URLs. We'd have to remember to repeat this test in all views that use this module. To reduce the duplication, you can place these common tests in a shared expectation, like this for example: https://github.com/18F/identity-idp/blob/master/spec/support/shared_examples_for_mailer.rb